doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,574 @@
+        &\text{HVFVerified}_i = \text{MAC}_{K_i}\text{(Identifier, ingress}_i\text{, egress}_i\text{, encryptedPolicyID}_i, \\& \text{srcAddrLen, srcHostAddr)[3:6] } \& \text{ 0x3FFFFF}\\\\
+    \end{align*}
+
+
+For accessing a sub slice we use the [a:b] notation, where we take the bytes from index a to index b, where b is excluded.
+For the DRKey notation, see :doc:`/cryptography/drkey`.
+The *srcAddrLen* and *srcHostAddr* are used as a MAC input for both the AS-Host DRKey case and the AS-AS DRKey case for simplicity.
+
+Data plane
+----------
+
+Processing at the router
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Whenever a FABRID enabled router receives a SCION packet, it has to figure out whether it should be processed as FABRID or not.
+If an AS receives a FABRID packet but does not support FABRID, it treats the packet as a normal SCION packet.
+In both cases, all the logic of a normal SCION packet will be applied too.
+The router determines whether the SCION packet is a FABRID packet as follows:
+
+.. image:: fig/FABRID/FABRIDActivation.png
+
+If the router supports FABRID and the SCION packet contains the FABRID HBH extension, the router is going to verify the
+correctness of the current FABRID Hop-validation-field using either the AS-to-AS or AS-to-Host DRKey and verifies whether
+the encrypted policy index matches a valid FABRID policy.
+If this is the case, the router will update the FABRID HVF to HVFVerified, see the :ref:`Header fields computation <fabrid-formulas>`,
+and route the packet over an intra-AS path matching the provided FABRID policy.
+All intra-AS paths are configured by the AS operator, and are provided to the border routers by the local control service.
+
+Processing at the endhost
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To be able to send a FABRID packet, the endhost has to choose a path that supports its path and policy constraints. A detailed explanation on how endhost applications can find such paths is given in the section :ref:`Exposing policies to the end hosts <endhost_policy_selection>`. Once a path has been found, with specific policies for each hop in the path, the path and an array containing one policy per hop is given to the FABRID snet implementation. The snet implementation then constructs the FABRID packet by automatically requesting the necessary DRKeys and computing the hop validation fields. The packet can then be sent to the border router for further forwarding. A receiving endhost can recompute the path validator to verify that the packet was forwarded over this path.
+With this the endhost is able to create FABRID packets and then send them to the border router for further forwarding.
+The FABRID snet implementation will automatically request the necessary DRKeys and compute the hop validation fields,
+the endhost only has to provide the path and the FABRID policies.
+Then the receiving endhost can recompute the path validator to verify that the packet was forwarded over this path.
+
+Control plane
+---------------
+
```

Comment:
Add brief overview?

Reply:
Would be nice to also have an overview of the most important interactions here, e.g., a few ASes each with some end host, explaining the different gRPC calls etc.